### DISCUSSION TITLE
Contributors Chat

### DISCUSSION THREAD

**Discussion by JeffSteinbok** [JeffSteinbok/hass-dreo#78] [General]
As opposed to using Discord or something like that, we can try this for chatting about PRs and such without overloading the issues that have been filed.

**Comment by JeffSteinbok** [comment]
@andy-cooper - We can try this instead of Discord for now.

**Reply by andy-cooper** [reply]
Perfect!

**Reply by andy-cooper** [reply]
So, what I was trying to do with my fork and branch off of my fork was to isolate the code for the oscillation angle on the wall-mounted heaters. When I pseudo-enable it on my space heaters, I see the controls to set the oscillation angle, but apparently the people who have a wall heater don't see it. It's not quite clear to me whether or not they actually used the oscangle branch on my fork or the main branch on my fork, so it's a little hard to tell what went wrong there. 

If things have been broken by my changes, I'll do my best to get that fixed rapidly.

**Reply by JeffSteinbok** [reply]
I'm updating my test VM now with the latest HA and I'll pull the code from Main and see if it works.  I don't have a space heater, but I do have a tower fan and an air circulator.  The latter supports angle setting.

**Reply by andy-cooper** [reply]
OK, great. One of these days I'll have to put together a test VM for myself. Right now I just run it against my live system. Hah. It does, however, provide motivation to fix it quickly.

**Reply by JeffSteinbok** [reply]
Still don't have the angle with what's in main at the moment.

Logger: homeassistant.components.number
Source: helpers/entity_platform.py:350
integration: Number (documentation, issues)
First occurred: 7:52:20 PM (1 occurrences)
Last logged: 7:52:20 PM

Error while setting up dreo platform for number
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 350, in _async_setup_platform
    await asyncio.shield(awaitable)
  File "/config/custom_components/dreo/number.py", line 101, in async_setup_entry
    async_add_entities(add_entries(manager.fans))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dreo/number.py", line 74, in add_entries
    if (de.device_definition.range[number_definition.attr_name+"_range"] is not None):
        ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'horizontal_angle_range'

**Reply by JeffSteinbok** [reply]
Back to 0.6.2 has no issues.

I saw a different exception in the HA CLI when I ran "core logs" but it scrolled up so much I couldn't see the start of it.

**Comment by andy-cooper** [comment]
OK, another PR to fix the KeyError - https://github.com/JeffSteinbok/hass-dreo/pull/79 - hopefully that will get your angles back. The risks of coding without a device and not properly expanding the unit tests.

**Comment by JeffSteinbok** [comment]
I am going to move b4 to a released build soon. Please let me know if any of you have other regressions?

@andy-cooper @iRonin @iRonin @crowbarz

**Comment by iRonin** [comment]
I can have a look at it tomorrow the earliest.

**Comment by heinzmerk** [comment]
I fixed an issue where for the humidifiers the humidity didn't update periodically. How can I share the updated code?

**Comment by talormanda** [comment]
Can't sign into PR 1.4.2

```
Logger: homeassistant.config_entries
Source: config_entries.py:761
First occurred: 9:10:20 PM (2 occurrences)
Last logged: 9:19:58 PM

Error setting up entry hidden_my_email_on_purpose@gmail.com for dreo
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 761, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dreo/__init__.py", line 60, in async_setup_entry
    load_devices = await hass.async_add_executor_job(pydreo_manager.load_devices)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 59, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/dreo/pydreo/__init__.py", line 236, in load_devices
    proc_return = self._process_devices(device_list)
  File "/config/custom_components/dreo/pydreo/__init__.py", line 202, in _process_devices
    self.load_device_state(device)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/config/custom_components/dreo/pydreo/__init__.py", line 271, in load_device_state
    device.update_state(device_state)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "/config/custom_components/dreo/pydreo/pydreohumidifier.py", line 197, in update_state
    self._rgblevel = RGB_MAP[self.get_state_update_value(state, RGB_LEVEL)]
                     ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: None
```

**Reply by JeffSteinbok** [reply]
Blech. Ok. I'll have a look after the ball game. 

Diagnostics pls. I make test cases from them.

**Reply by talormanda** [reply]
<img width="1169" height="794" alt="image" src="https://github.com/user-attachments/assets/ac269c54-2682-493e-9bb3-3f7bf1e78636" />

I don't have any option to do that from the integration.

**Reply by talormanda** [reply]
[home-assistant_2025-10-06T01-49-21.968Z.log](https://github.com/user-attachments/files/22713551/home-assistant_2025-10-06T01-49-21.968Z.log)

Here is a HA one. The Dreo link doesn't show up since it can't authenticate. If I do migrate to 1.3.3 it works fine.

**Comment by talormanda** [comment]
Side conversation: I just got a new Humidifier: DR-HHM014S

I can share the device or add to a new account and give you the login to help get this added. Is this possible? Right now I only have basic controls. No power controls.

<img width="666" height="234" alt="image" src="https://github.com/user-attachments/assets/555802af-aff4-4fdc-a82f-ee44c982bac4" />

**Comment by JeffSteinbok** [comment]
Open a new Issue please.  Diagnostics and debug logs when you try to operate it.
Anything stating with HHM though **should** get you the basic humidifier entity.
What version are you on?

**Reply by talormanda** [reply]
I have stable 1.3.3 on my main and a test HA instance where I can install whatever.

**Comment by JeffSteinbok** [comment]
Also, the issue doesn't appear to be authentication.  But it's crapping out so early, I don't get the logs.

This line is crapping out because "none" is the value passed into the array.
self._rgblevel = RGB_MAP[self.get_state_update_value(state, RGB_LEVEL)].  I'll put some quick error checking around it and push to main so hopefully it unblocks.

**Comment by JeffSteinbok** [comment]
Just pushed a quick safety fix to main, can you pull from there?
Hopefully the entity loads and we can get better logs.

I'm on discord now if you are "@jeffstei"
